src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1618,13 +1626,17 @@ def close_libvirt_connection(self):
         try:
             if not await self.middleware.call('service.started', 'libvirtd'):
                 await asyncio.wait_for(libvirtd_started(self.middleware), timeout=timeout)
+            # We want to do this before initializing libvirt connection
```

Comment:
I just noticed above, unlreated to this change: time.sleep(2), thats blocks the event loop

use await asyncio.sleep

We should also have a limit for waiting